railgun/internal/diagnostics/server.go

```diff
@@ -46,8 +63,28 @@ func (s *Server) Listen(ctx context.Context) error {
 	}
 }
 
-// installHandlers adds the Profiling webservice to the given mux.
-func installHandlers(mux *http.ServeMux) {
+// receiveConfig watches the config update channel
+func (s *Server) receiveConfig(ctx context.Context) {
+	for {
+		select {
+		case dump := <-s.ConfigDumps.Configs:
+			if !dump.Failed {
+				successfulConfigDump = dump.Config
+			} else {
+				failedConfigDump = dump.Config
+			}
+		case <-ctx.Done():
+			if err := ctx.Err(); err != nil {
+				s.Logger.Error(err, "error received while stopping diagnostic config collection")
+			}
+			s.Logger.V(3).Info("shutting down diagnostic config collection")
```

Comment:
Good logging :+1: 
Only small nits:

```suggestion
			if err := ctx.Err(); err != nil {
				s.Logger.Error(err, "shutting down diagnostic config collection: context completed with error")
			}
			s.Logger.V(3).Info("shutting down diagnostic config collection: context completed")
```

I can see an argument that language about go contexts in operator logs might be a bit of a leaky abstraction, but I think the benefits outweigh the problems.
I'm also open to changing the language to something that explains the context, such as:

```go
			if err := ctx.Err(); err != nil {
				s.Logger.Error(err, "diagnostic server stopping due to signal from main program. an error occurred.")
			}
			s.Logger.V(3).Info("diagnostic server stopping due to signal from main program.")
```

Think about it, but note that I don't consider this comment blocking as we don't really have a standard written up for this in our codebases atm. Resolve at your discretion.